WordPress/gutenberg
# Font Library: add option to revoke access to Google Fonts

## What?

Add an option to revoke access to Google Fonts to the the Font Library modal.

Partial for https://github.com/WordPress/gutenberg/issues/54826
(Partial because it only supports Google Fonts not other third parties).

## Why?

There was no option for revoking access to Google Fonts once access was granted.

## How?

Adds a dropdown menu with an option that sets the 'wp-font-library-google-fonts-permission' local storage to false.

## Testing Instructions




Activate a block theme.
Open the styles sidebar in the Site Editor.
Open the Typography panel, then open the Font Library
Select the "Install Fonts" tab.

If you have not already granted permission to access Google Fonts, there should be no visible change when the PR is applied.

After allowing access to Google Fonts, there should be a button with an ellipsis to the right of the "Google Fonts" title.
The ellipsis should not show when you view a single font. Only on the first page and on the paged results.
Clicking the option should open a popup with a single option "Revoke access to Google Fonts".
Selecting this option should display the "Connect to Google Fonts" banner and button again.

### Testing Instructions for Keyboard


## Screenshots or screencast 

<img width="976" alt="Font Library modal with the new ellipsis menu with the Revoke access option." src="https://github.com/WordPress/gutenberg/assets/7422055/db601eb5-33d8-4723-acf8-6e9233d8e4c4">
